fwoa-tools/src/binaryConverter.ts

```diff
@@ -143,32 +152,31 @@ async function checkConfiguration(): Promise<void> {
 }
 
 async function startBinaryConversion(): Promise<void> {
-  for (const folderName of Object.keys(outputFile.file_names)) {
-    console.log(`Starting Binary Conversion for folder ${folderName}`);
-    await retrieveBinaryIdsFromFolder(folderName);
-    console.log(`Finished Binary Conversion for folder ${folderName}`);
+  if (!outputFile.file_names.Binary) {
+    logs.write(`${new Date().toISOString()}: No Binary resources found to convert! Exiting...\n`);
+    return;
   }
+  console.log(`Starting Binary Resource Conversion...`);
+  logs.write(`${new Date().toISOString()}: Starting Binary Resource Conversion`);
+  await convertBinaryResource();
+  logs.write(`${new Date().toISOString()}: Finished Binary Resource Conversion`);
 }
 
-if (!dryRun) {
-  startBinaryConversion()
-    .then(() => {
+(async () => {
+  await checkConfiguration();
+  console.log('successfully authenticated to all services');
+  if (!dryRun) {
+    try {
+      await startBinaryConversion();
       console.log('successfully converted all binary resources!');
-      // eslint-disable-next-line security/detect-non-literal-fs-filename
-      writeFileSync(`${CONVERSION_OUTPUT_LOG_FILE_PREFIX}${Date.now().toString()}.log`, logs.join('\n'));
-    })
-    .catch((error) => {
+      logs.end();
```

Comment:
I think you can move this outside of the `if statement` and to line 178. In that case if there are any logs written it `checkConfiguration`, it'll be captured as well.

Reply:
Updated 👍